charts/replicated-library/templates/lib/_service_headless.tpl

```diff
@@ -0,0 +1,20 @@
+{{/*
+Renders the Service objects required by the chart.
+*/}}
+{{- define "replicated-library.service_headless" -}}
+  {{- /* Generate named services as required */ -}}
+  {{- range $name, $service := .Values.services }}
+    {{- if $service.headless -}}
+      {{- $serviceValues := $service -}}
+
+      {{- $_ := set $ "ObjectName" $name -}}
+      {{- if $serviceValues.nameOverride -}}
+        {{- $_ := set $ "ObjectName" $serviceValues.nameOverride -}}
+      {{ end -}}
+
+      {{- $_ := set $ "ObjectValues" (dict "service" $serviceValues) -}}
+
+      {{- include "replicated-library.classes.service_headless" $ | nindent 0 }}
+    {{- end }}
+  {{- end }}
+{{- end }}
```

Comment:
@drohnow I'm curious why we've gone the direction of adding a separate template for headless services? With how you've written this now, this will create a duplicate headless service for each service that's defined in `.Values.services`. 

If the only different is between a regular service and a headless one is when `clusterIP` is set to false, I think we're okay to skip adding a separate template and just document that in the `values.yaml`